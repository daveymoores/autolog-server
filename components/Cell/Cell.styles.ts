import css from "styled-jsx/css";

export default css`
  .timesheet__cell {
    margin-top: calc(-1 * var(--lineWidth));
    padding: 0.82em 1.2em;
    font-size: 16px;
    text-align: center;
  }

  .timesheet__cell:first-child {
    margin-top: 0;
  }

  .timesheet__cell p {
    padding: 0;
    margin: 0;
  }

  .timesheet__cell--title {
    font-weight: 600;
  }

  @media (prefers-color-scheme: dark) {
    .timesheet__cell {
      border: var(--lineWidth) solid var(--lightGreen);
    }

    .timesheet__cell--title {
      background-color: var(--mistyGrey);
    }
  }

  @media (prefers-color-scheme: light) {
    .timesheet__cell {
      border: var(--lineWidth) solid var(--darkGrey);
    }

    .timesheet__cell--title {
      background-color: var(--mistyGreen);
    }
  }
`;
