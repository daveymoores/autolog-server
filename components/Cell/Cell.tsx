import React from "react";

import styles from "./Cell.styles";

interface CellProps {
  text: string;
  title?: boolean;
}

const Cell: React.FC<CellProps> = ({ text, title }) => {
  return (
    <>
      <div
        className={`timesheet__cell${title ? " timesheet__cell--title" : ""}`}
      >
        <p>{text}</p>
      </div>
      <style jsx>{styles}</style>
    </>
  );
};

export default Cell;
