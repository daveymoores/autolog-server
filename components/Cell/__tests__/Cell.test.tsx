import React from "react";
import renderer from "react-test-renderer";

import Cell from "../Cell";

describe("Cell", () => {
  it("renders Cell unchanged", () => {
    const tree = renderer.create(<Cell text={"Hello"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
