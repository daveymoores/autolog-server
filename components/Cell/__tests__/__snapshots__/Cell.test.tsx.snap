// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell renders Cell unchanged 1`] = `
<div
  className="jsx-16556622 timesheet__cell"
>
  <p
    className="jsx-16556622"
  >
    Hello
  </p>
</div>
`;
